### ISSUE TITLE
NPC infinite movement loop

### ISSUE THREAD

**Issue by Drakasin** [CleverRaven/Cataclysm-DDA#26237]
**Describe the bug**  
NPC stuck in infinite movement loop while zombie attacks him

**To Reproduce**  
Gotta be (un)lucky to reproduce I guess.

**Expected behavior**  
The NPC actually moves and strikes back etc.

**Screenshots**  
https://i.imgur.com/lc6dNEz.png  (the NPC is the one on screen)

**Versions and configuration(please complete the following information):**
 - OS: W10 64bit
 - Game Version: 0.C-33350-g089bb94
 - Graphics version: Tiles
 - Mods loaded: DDA, simplified nutrition, Disable NPC Needs, filthy clothing, NPC mutations, NPC traits

**Additional context**  
NPC halted me, I convinced him not to kill me, I trade him booze for lighter, he runs off then gets followed by a fat zombie. He gets stuck in a movement loop moving left and right, left and right, while the zombie attacks him. He doesn't strike back. When I pressed space to continue he died immediately.

**Comment by FulcrumA** [comment]
Pathfinding etc in general is still pretty broken for NPCs. Similar with the issue of an NPC running off, running into a wall then staying there for years staring at that wall unless something kills it or the wall gets torn down.

**Comment by Karthas077** [comment]
Possibly related, but all of the spawn point NPCs I've generated are unwilling to attack a Fat Zombie unless I am also standing next to the zombie. This includes a ridiculously strong NPC who happened to have a glock with approximately 30 bullets. I baited a horde into that guy and he killed every zombie that even got close with one shot until a fat zombie got to him and then he just stood there and let himself be eaten.
